examples/ui/display_and_visibility.rs

```diff
@@ -0,0 +1,407 @@
+//! Demonstrates how Display and Visibility work in the UI.
+
+use bevy::prelude::*;
+use bevy::winit::WinitSettings;
+
+const PALETTE: [&str; 4] = ["4059AD", "6B9AC4", "A5C8E1", "F4B942"];
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        // Only run the app when there is user input. This will significantly reduce CPU/GPU use.
+        .insert_resource(WinitSettings::desktop_app())
+        .add_startup_system(setup)
+        .add_system(buttons_handler::<Display>)
+        .add_system(buttons_handler::<Visibility>)
+        .add_system(text_hover)
+        .run();
+}
+
+#[derive(Component)]
+struct Target<T> {
+    id: Entity,
+    phantom: std::marker::PhantomData<T>,
+}
+
+impl<T> Target<T> {
+    fn new(id: Entity) -> Self {
+        Self {
+            id,
+            phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+trait TargetUpdate {
+    type TargetComponent: Component;
+    const NAME: &'static str;
+    fn update_target(&self, target: &mut Self::TargetComponent) -> String;
+}
+
+impl TargetUpdate for Target<Display> {
+    type TargetComponent = Style;
+    const NAME: &'static str = "Display";
+    fn update_target(&self, style: &mut Self::TargetComponent) -> String {
+        style.display = match style.display {
+            Display::Flex => Display::None,
+            Display::None => Display::Flex,
+        };
+        format!(" {}::{:?} ", Self::NAME, style.display)
+    }
+}
+
+impl TargetUpdate for Target<Visibility> {
+    type TargetComponent = Visibility;
+    const NAME: &'static str = "Visibility";
+    fn update_target(&self, visibility: &mut Self::TargetComponent) -> String {
+        *visibility = match *visibility {
+            Visibility::Inherited => Visibility::Visible,
+            Visibility::Visible => Visibility::Hidden,
+            Visibility::Hidden => Visibility::Inherited,
+        };
+        format!(" {}::{visibility:?}", Self::NAME)
```

Comment:
I just noticed inconsistent use of spaces in `49`, `62` and `369`. I would either:

- ` AAA `: space around
- `AAA`: no spaces

Reply:
I've no idea why it was this way, deleted the spaces as you suggested.